File: app/src/main/java/com/example/todomanagement/MainActivity.java
/*
This is our MainActivity. It is the first screen you see when you open this application. It shows a List of all
Tasks that you have created. From here on you can add new tasks. You can check a task if it is complete or
see further details when you click on it.
 */

package com.example.todomanagement;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.CheckBox;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements RecyclerAdapter.OnTaskListener {
    public static final String Extra_ID = "com.example.todomanagement.ID";
    public static final String Extra_TITLE = "com.example.todomanagement.TITLE";
    public static final String Extra_DESC = "com.example.todomanagement.DESC";
    public static final String Extra_DATE = "com.example.todomanagement.DATE";
    public static final String Extra_STATUS = "com.example.todomanagement.STATUS";
    private final DatabaseHandler db = new DatabaseHandler(this);
    private RecyclerView recyclerView;
    private ArrayList<Task> tasks;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FloatingActionButton button = findViewById(R.id.button_new_task);
        recyclerView = findViewById(R.id.recyclerView);

        db.openDatabase();
        tasks = db.getAllTasks();

        setAdapter();

        button.setOnClickListener(view -> {
            Intent intent = new Intent(getApplicationContext(), AddTaskActivity.class);
            startActivity(intent);
        });
    }

    //Sets the adapter of the recyclerView
    public void setAdapter() {
        RecyclerAdapter adapter = new RecyclerAdapter(this,this);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onTaskClick(int position) {
        Intent intent = new Intent(this, TaskDetailsActivity.class);
        String id = String.valueOf(tasks.get(position).getId());
        String title = tasks.get(position).getTitle();
        String desc = tasks.get(position).getDescription() + " ";
        String date = tasks.get(position).getDueDate();
        //String status = String.valueOf(tasks.get(position).getStatus());
        CheckBox checkbox = (CheckBox) findViewById(R.id.checkbox_view);

        String status = String.valueOf(checkbox.isChecked());
        Log.d("Status..", status);

        intent.putExtra(Extra_ID, id);
        intent.putExtra(Extra_TITLE, title);
        intent.putExtra(Extra_DESC, desc);
        intent.putExtra(Extra_DATE, date);
        intent.putExtra(Extra_STATUS, status);

        startActivity(intent);
    }
}

File: app/src/main/java/com/example/todomanagement/MainActivity.kt
package com.example.todomanagement

import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.drawable.Drawable
import android.os.Build
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import android.view.Menu
import android.view.MenuItem
import android.widget.EditText
import android.widget.TextView
import androidx.core.graphics.drawable.DrawableCompat
import androidx.core.view.ViewCompat
import androidx.core.view.ViewCompat.setBackgroundTintList
import com.example.todomanagement.databinding.ActivityMainBinding
import com.example.todomanagement.databinding.NewTaskActivityBinding
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.*
import kotlin.time.measureTimedValue

class MainActivity : AppCompatActivity() {

    private lateinit var taskmanager: TaskManager
    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding
    private lateinit var tas_binding: NewTaskActivityBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.toolbar)

        taskmanager = TaskManager()

        tas_binding = NewTaskActivityBinding.inflate(layoutInflater)

        binding.buttonNewTask.setOnClickListener { view -> openNewTaskDisplay()}
        //tas_binding.buttonAddNewTask.setOnClickListener{ view -> doNowSth()}
    }

    fun showAllTasks(){
        taskmanager.showAllTasks()
    }

    fun openNewTaskDisplay(){
        setContentView(R.layout.new_task_activity)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when(item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
    val navController = findNavController(R.id.nav_host_fragment_content_main)
    return navController.navigateUp(appBarConfiguration)
            || super.onSupportNavigateUp()
    }

    fun addNewTask(view: android.view.View) {
        val task_title: TextView = findViewById(R.id.task_title)
        val task_dueDate: TextView = findViewById(R.id.task_due_date)

        val title: String = task_title.text.toString()
        val dueDate: String = task_dueDate.text.toString()
        val description: String = tas_binding.taskDescription.toString()

        if (title.length == 0 || dueDate.length == 0){
            checkTaskDetails(task_title, task_dueDate)
            return
        }

        val task: Task = taskmanager.createNewTask(title, description, dueDate)
        taskmanager.addNewTask(task)

        showAllTasks()
        setContentView(binding.root)
    }

    fun checkTaskDetails(task_title: TextView, task_dueDate: TextView){

        val title = task_title.text.toString()
        val dueDate = task_dueDate.text.toString()

        if(title.length == 0) {
            ViewCompat.setBackgroundTintList(
                task_title,
                ColorStateList.valueOf(Color.parseColor("#FF0000")));
        } else {
            ViewCompat.setBackgroundTintList(
                task_title,
                ColorStateList.valueOf(Color.parseColor("#808080")));
        }

        if(dueDate.length == 0) {
            ViewCompat.setBackgroundTintList(
                task_dueDate,
                ColorStateList.valueOf(Color.parseColor("#FF0000")));
        } else {
            ViewCompat.setBackgroundTintList(
                task_dueDate,
                ColorStateList.valueOf(Color.parseColor("#808080")));
        }

    }
}